climada/hazard/centroids/centr.py

```diff
@@ -793,9 +781,12 @@ def remove_duplicate_points(self, scheduler=None):
         """
         if not self.lat.any() and not self.meta:
             return self
-        self.set_geometry_points(scheduler)
-        geom_wkb = self.geometry.apply(lambda geom: geom.wkb)
-        sel_cen = geom_wkb.drop_duplicates().index
+        if self.lat.size > 0:
+            coords_view = self.coord.astype(np.float64).view(dtype='float64,float64')
+            sel_cen = np.sort(np.unique(coords_view, return_index=True)[1])
```

Comment:
If the points are stored in `self.lat` and `self.lon`, it's much faster to use these two lines to identify duplicate points than applying `set_geometry_points` and using `drop_duplicates` afterwards.

Reply:
If both `self.lat` and `self.geometry` are defined, which one is faster?

Reply:
It's definitely much faster to use pure NumPy functionality wherever possible. In this case, the operation `self.geometry.apply(lambda geom: geom.wkb)` alone is already factor 10 slower than the two lines above. The `geom_wkb.drop_duplicates().index` operation then doesn't make much of a difference.